src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -118,6 +119,36 @@ object Instance {
     def apply(runSpec: RunSpec): Instance = Scheduled(runSpec, Id.forRunSpec(runSpec.id))
   }
 
+  object Provisioned {
+    /**
+      * Factory method for creating provisioned instance from Scheduled instance for apps
+      * @param scheduledInstance instance in a Scheduled state
+      * @return new instance in a provisioned state
+      */
+    def apply(scheduledInstance: Instance, agentInfo: AgentInfo, networkInfo: core.task.state.NetworkInfo, app: AppDefinition)(implicit clock: Clock): Instance = {
```

Comment:
Why not simply pass `now` as a value?

Reply:
ok ... I had that and then switched to clock, but maybe timestamp actually makes more sense :)